cmake/mason-dependencies.cmake

```diff
@@ -0,0 +1,17 @@
+mason_use(geometry VERSION 0.9.2 HEADER_ONLY)
+mason_use(variant VERSION 1.1.4 HEADER_ONLY)
+mason_use(unique_resource VERSION cba309e HEADER_ONLY)
+mason_use(rapidjson VERSION 1.1.0 HEADER_ONLY)
+mason_use(boost VERSION 1.65.1 HEADER_ONLY)
+mason_use(geojsonvt VERSION 6.3.0 HEADER_ONLY)
+mason_use(supercluster VERSION 0.2.2 HEADER_ONLY)
+mason_use(kdbush VERSION 0.1.1-1 HEADER_ONLY)
+mason_use(earcut VERSION 0.12.4 HEADER_ONLY)
+mason_use(protozero VERSION 1.5.2 HEADER_ONLY)
+mason_use(pixelmatch VERSION 0.10.0 HEADER_ONLY)
+mason_use(geojson VERSION 0.4.2 HEADER_ONLY)
+mason_use(polylabel VERSION 1.0.3 HEADER_ONLY)
+mason_use(wagyu VERSION 0.4.3 HEADER_ONLY)
+mason_use(shelf-pack VERSION 2.1.1 HEADER_ONLY)
+mason_use(cheap-ruler VERSION 2.5.3 HEADER_ONLY)
+mason_use(vector-tile VERSION 1.0.1 HEADER_ONLY)
```

Comment:
This only tracks some of the dependencies. All platforms add additional dependencies with `mason_use`. E.g. Linux adds https://github.com/mapbox/mapbox-gl-native/blob/c1f16afdd1e4838b3ec15a7e057025646f1014b6/platform/linux/config.cmake#L1-L11

Reply:
https://github.com/mapbox/mapbox-gl-native/pull/12328/commits/5dcd8e8d5e3b3001c580370cb4bfcf9dc16bd2db. On the fence about this, WDYT?